avocado/core/multiplexer.py

```diff
@@ -405,25 +410,30 @@ def parse(self, args):
         self.data_inject = _report_mux_already_parsed
         self.data_merge = _report_mux_already_parsed
 
-    def _parse_basic_injects(self, args):
+    def _process_default_params(self, args):
         """
-        Inject data from the basic injects defined by Mux
+        Process the default params
 
         :param args: Parsed cmdline arguments
         """
+        default_params = tree.TreeNode()
+        for default_param in self.default_params.itervalues():
+            default_params.merge(default_param)
+        self.default_params = default_params
         # FIXME: Backward compatibility params, to be removed when 36 LTS is
         # discontinued
         if (not getattr(args, "mux_skip_defaults", False) and
                 hasattr(args, "default_avocado_params")):
-            self.data_merge(args.default_avocado_params)
+            self.default_params.merge(args.default_avocado_params)
+        return self.default_params
 
     def is_parsed(self):
         """
         Reports whether the tree was already multiplexed
         """
         return self.variants is not None
 
-    def data_inject(self, key, value, path=None):
+    def data_inject(self, name, key, value, path=None):
```

Comment:
In this version it should only identify them, but later I plan to add support for filters.